### PR TITLE
Feature/us8382 pledge campaign and general fund

### PR DESCRIPTION
**Pull request by silbermm**

### CODE REVIEW

File: CI/SQL/20170614_151701_US8382-ConfigurationSettingsForTripPledgeCampaign.sql [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		 [Application_Code]
+		,[Key_Name]
+		,[Value]
+		,[Domain_ID]
+		,[Description]
+	) VALUES (
+		 @APP_CODE
+		,'TripApplication'
+		,'20'
+		,1
```
Comment (andycanterbury) [inline CI/SQL/20170614_151701_US8382-ConfigurationSettingsForTripPledgeCampaign.sql]: Could have put all these domain id's in a variable....

File: MinistryPlatform/Models/MpEvent.cs [hunk 2]
```diff
@@ -46,4 +46,13 @@ public class MpEvent
         [JsonProperty(PropertyName = "On_Donation_Batch_Tool")]
         public bool OnDonationBatchTool { get; set; }
     }
+
+    public class MpSimpleEvent
```
Comment (andycanterbury) [inline MinistryPlatform/Models/MpEvent.cs]: Do we need two classes for events? We typically put one class per file also.

File: crds-mp-tools/Services/Trips/TripSetupService.cs [hunk 3]
```diff
@@ -33,25 +44,84 @@ public IObservable<List<StaffContact>> GetStaffContacts()
...
         }
 
+        public IObservable<MpPledge> CreatePledgeAndCampaign(TripSetupViewModel tripData, MpSimpleEvent mpSimpleEvent)
+        {
+
+            try
+            {
+                var pledgeCampaign = GetPledgeCampaign(tripData, mpSimpleEvent);
+
+                return _pledgeCampaignRepository.Create(pledgeCampaign).Select(campaign =>
```
Comment (andycanterbury) [inline crds-mp-tools/Services/Trips/TripSetupService.cs]: I know this is the functional style, but it seems weird to me to have a return statement that goes off and does a whole bunch more stuff, including creating records. Couldn't you take all the stuff inlined here and put it into a "normal" function that could be reused?

File: crds-mp-tools/Services/Trips/TripSetupService.cs [hunk 4]
```diff
@@ -79,14 +149,47 @@ private MpEvent GetEvent(TripSetupViewModel tripData, MpProgram program)
...
+                EventId = simple.EventId,
+                ProgramId = simple.ProgramId
+            };
+        }
+
+        private MpPledge GetPledge(int pledgeCampaignId)
+        {
+            return new MpPledge
+            {
+                DonorId = _configWrapper.GetMpConfigIntValue(AppCode, "TripGeneralFundDonor") ?? 7726454,
```
Comment (andycanterbury) [inline crds-mp-tools/Services/Trips/TripSetupService.cs]: Magic number? Should probably go in config settings.

File: crds-mp-tools/Services/Trips/TripSetupService.cs [hunk 5]
```diff
@@ -79,14 +149,47 @@ private MpEvent GetEvent(TripSetupViewModel tripData, MpProgram program)
...
+                PledgeCampaignId = pledgeCampaignId,
+                PledgeStatusId = _configWrapper.GetMpConfigIntValue(AppCode, "ActivePledgeStatus") ?? 1
```
Comment (andycanterbury) [inline crds-mp-tools/Services/Trips/TripSetupService.cs]: Magic number? Should probably go in config settings.

File: MinistryPlatform/Models/MpEvent.cs (hunk 2 above)
Comment (Ganonside) [inline MinistryPlatform/Models/MpEvent.cs]: This class lets us to grab the returning ID's from the returning database calls.

File: crds-mp-tools/Services/Trips/TripSetupService.cs (hunk 4 above)
Comment (silbermm) [inline crds-mp-tools/Services/Trips/TripSetupService.cs]: It is a config value. This is the same thing we did with Events and Programs. We try to get the value out of the MpConfig table and if it doesn't exist we use the default (which is currently the same thing stored in the table.).

File: crds-mp-tools/Services/Trips/TripSetupService.cs (hunk 5 above)
Comment (silbermm) [inline crds-mp-tools/Services/Trips/TripSetupService.cs]: It is a config value. This is the same thing we did with Events and Programs. We try to get the value out of the MpConfig table and if it doesn't exist we use the default (which is currently the same thing stored in the table.).

File: crds-mp-tools/Services/Trips/TripSetupService.cs (hunk 4 above)
Comment (andycanterbury) [inline crds-mp-tools/Services/Trips/TripSetupService.cs]: can we put the I'm an Idiot tag in comments?

**Pull request merged**